lib/promauth/config.go

```diff
@@ -473,6 +479,20 @@ func (ac *Config) NewTLSConfig() (*tls.Config, error) {
 	return tlsCfg, nil
 }
 
+// NewRoundTripper returns new http.RoundTripper for the given ac.
+func (ac *Config) NewRoundTripper(newRT func(*tls.Config) (http.RoundTripper, error)) (http.RoundTripper, error) {
```

Comment:
I don't really like this API. 

From user perspective, maybe it's better to have an option to override default transport values?

E.g. 

```go
func (ac *Config) NewRoundTripperWithBuilder(builder func(*http.Transport)) (http.RoundTripper, error) {
   ...
}
```

Usage:

```go
rt, err := sw.AuthConfig.NewRoundTripperWithBuilder(func(tr  *http.Transport)  {
			tr.Proxy =   proxyURLFunc,
			tr.TLSHandshakeTimeout =   10 * time.Second
			tr.IdleConnTimeout = 2 * sw.ScrapeInterval
	})
```

Reply:
I like the idea, implemented this.